src/core/math/matrixZipMap.ts

```diff
@@ -2,70 +2,109 @@ import * as R from 'ramda';
 
 type Matrix<T> = T[][];
 
-export function matrixMap<T1, T2, TR>(
+export function matrixMap<T1, TR>(
     m1: Matrix<T1>,
-    m2: Matrix<T2>,
-    cb: (d1: T1, d2: T2, i: number, j: number) => TR
+    cb: (d1: T1, i: number, j: number) => TR
 ) {
     const mapMatrix = R.addIndex<T1[], TR[]>(R.map);
     const mapRow = R.addIndex<T1, TR>(R.map);
 
     return mapMatrix((iRow, i) =>
         mapRow(
-            (ijValue, j) => cb(ijValue, m2[i][j], i, j),
+            (ijValue, j) => cb(ijValue, i, j),
+            iRow
+        ), m1
+    );
+}
```

Comment:
Adding a 1-matrix map -- otherwise all below is essentially dealing with undefined cases